### DISCUSSION TITLE
Adding a reply markup to a channel post remove comment / discussion option

### DISCUSSION THREAD

**Discussion by elBenoCapo** [python-telegram-bot/python-telegram-bot#2400] [Q&A]
Hello everyone,

I'm currently making a bot that send messages to a public channel. I'd also like that people can comment under those posts, so I activated the comment option
All was working perfectly ( [screenshot](https://i.imgur.com/f4ggMqE.png) ) until I started messing with the reply markup to add emoji reactions 
```python            
context.bot.send_photo(chat_id=channel,
       photo=image,
       caption=caption,
       parse_mode="html",
       reply_markup=initial_button_list_chart
       )
```
where `initial_button_list_chart` is an `InlineKeyboardMarkup`
Since I added that, I now have the reply_markup that I wanted ( [screenshot](https://i.imgur.com/8bVPSMy.png) ), but the comment button no longer appears.

Similar issue happen when using `send_message` rather than `send_photo`

I tried looking at the doc and online, but couldn't find a solution / existing similar issues.

Is there a way to add a reply_markup while keeping the comment feature?

Thanks in advance,
Ben

**Comment by Poolitzer** [comment]
> Is there a way to add a reply_markup while keeping the comment feature?

No, there is not. You would need to contact BotSupport to ask them to implement a way.

**Reply by elBenoCapo** [reply]
Thanks a lot for your quick answer. I'll try to contact them.

**Reply by Poolitzer** [reply]
good luck

**Comment by mrienstra** [comment]
Related conversations:

**2024: https://bugs.telegram.org/c/41803**
> This is the expected behavior at the moment. It may change in future
> 
> In channels where the comment group is connected, if a post with an inline keyboard button is sent, the comments will not be displayed in the channel, and comments can be sent to that post only through the membership of the comment group. .
> 
> Steps to reproduce
> 1- Create a channel and consider a comment group for it.
> 2- Send a post with an inline keyboard in the channel.
> 
> Current Result
> Comments are not displayed on the channel for that post.

**2020: https://github.com/python-telegram-bot/python-telegram-bot/issues/2248**

Suggests some work-arounds, e.g. "adding an URL inline button which has the text "comment" and then link to the specific message in the discussion group, it being the post of the channel."
